File: vane-portals/src/main/java/org/oddlama/vane/portals/event/PortalConstructEvent.java
package org.oddlama.vane.portals.event;

import org.bukkit.entity.Player;
import org.bukkit.event.HandlerList;
import org.oddlama.vane.portals.portal.PortalBoundary;

public class PortalConstructEvent extends PortalEvent {

    private static final HandlerList handlers = new HandlerList();
    private Player player;
    private PortalBoundary boundary;
    private boolean checkOnly;

    public PortalConstructEvent(final Player player, final PortalBoundary boundary, boolean checkOnly) {
        this.player = player;
        this.boundary = boundary;
        this.checkOnly = checkOnly;
    }

    public Player getPlayer() {
        return player;
    }

    public PortalBoundary getBoundary() {
        return boundary;
    }

    public boolean checkOnly() {
        return checkOnly;
    }

    public HandlerList getHandlers() {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: vane-portals/src/main/kotlin/org/oddlama/vane/portals/event/PortalConstructEvent.kt
package org.oddlama.vane.portals.event

import org.bukkit.entity.Player
import org.bukkit.event.HandlerList
import org.oddlama.vane.portals.portal.PortalBoundary

/**
 * Fired while validating or constructing a new portal from a boundary.
 *
 * @property player the player initiating construction.
 * @property boundary the detected boundary.
 * @property checkOnly whether this is a dry-run validation.
 */
class PortalConstructEvent(@JvmField val player: Player?, @JvmField val boundary: PortalBoundary?, private val checkOnly: Boolean) :
    PortalEvent() {
    /** Returns whether this event is in validation-only mode. */
    fun checkOnly() = checkOnly

    /** Returns Bukkit handlers for this event. */
    override fun getHandlers() = handlerList

    /** Static Bukkit handler list for this event type. */
    companion object {
        /** Shared handler list used by Bukkit's event system. */
        @JvmStatic val handlerList: HandlerList = HandlerList()
    }
}
